### ISSUE TITLE
AppImage: Icon resolution

### ISSUE THREAD

**Issue by rebio** [Beep6581/RawTherapee#6527]
**Short description**
The resolution of the AppImage icon varies.

**Steps to reproduce**
Can not be reproduced within the program

![RT 3101, 3102, 3104](https://user-images.githubusercontent.com/10911084/180594945-05d204f6-a2fa-4b38-b802-81965e20f0ec.png)

**Expected behavior**
Icon looks the same every time

**Additional information**
AppImage used on KDE neon

The displayed resolution of the logo was way more pixelated in the past (like 8 x 8px). Then it got way better as it is shown for 3101. Now it get's worse.

I don't care much about the logo resolution. If you are not Captain Care'a'lot either feel free to close the report.
I wanted to mention it as soon as I noticed it to avoid the unanswerable question which commit the change was caused by.

**Comment by Lawrence37** [comment]
This could be because of the desktop environment. I downloaded the 3101 AppImage and the icon is 32x32, the same as the one you show for 3102.

**Comment by rebio** [comment]
I tested this on other DEs as well:
- ubuntu-22.04-desktop-amd64.iso (Ubuntu Gnome)
- neon-user-20220721-0952.iso (default KDE)
- elementaryos-6.1-stable.20211218-rc.iso (Pantheon)
- Fedora-Workstation-Live-x86_64-36-1.5.iso (default Gnome)
- manjaro-kde-21.3.5-220721-linux515.iso (KDE with manjaro branding)
- MX-21.1_KDE_x64.iso (MX KDE)

All tested DEs except one only showed a generic AppImage icon and not the RT icon. The one that did show the icon was KDE neon with default KDE (neon-user-20220721-0952.iso). So exactly what I use. Neither Gnome nor Pantheon displayed the icons. The strange thing: Other distros with KDE did not either.

I ran all systems from a live USB. I wouldn't expect that to be any different but I don't know for sure.

**Comment by rebio** [comment]
A bug report at the KDE bugtracker was closed as this problem is caused by the AppImage: https://bugs.kde.org/show_bug.cgi?id=456970

**Comment by Lawrence37** [comment]
So they use the .DirIcon. The AppImage build process uses linuxdeploy to automatically generate that file. Unless there is an argument for using a specific resolution, I will leave it up to linuxdeploy to choose.

**Comment by rebio** [comment]
I don't know if anybody has changed the code for this, but 3151-ge5007c77 has a higher resoluted icon again

![RT icons](https://user-images.githubusercontent.com/10911084/192687598-39ca123f-03ba-436c-960a-7106dee8b434.png)

**Comment by Thanatomanic** [comment]
Thanks for the follow-up. So, the report is no longer valid?

**Comment by rebio** [comment]
Considering that the quality was changing in the past and switched through various resolutions I would wait until the next Automated build to really say more about this. If there was no change for a component regarding the building process or how the logo is handled I wouldn't expect that the random nature of this changed.
Considering the pattern I observed, the chosen logo resolution seems to be somewhat random for each build. (3101 & 3151 = High resolution, 3102 = medium resolution, 3104 - 3140 = low resolution)

I noted it as soon as I observed it so it's easier to find the bug.
